pkgs/applications/version-management/sourcehut/core.nix

```diff
@@ -0,0 +1,75 @@
+{ stdenv, fetchgit, fetchNodeModules, buildPythonPackage
+, flask, flask_login, humanize, sqlalchemy, sqlalchemy-utils, psycopg2, markdown
+, bleach, requests, beautifulsoup4, pgpy, misaka, pygments, alembic
+, sassc, nodejs-11_x
+, writeText }:
+
+buildPythonPackage rec {
+  pname = "srht";
+  version = "0.30.8";
+
+  src = fetchgit {
+    url = "https://git.sr.ht/~sircmpwn/core.sr.ht";
+    rev = version;
+    sha256 = "17rfcd5xfhkhcj5n906nrr2i00p8kg1a1v5wfc1ysywp3kzhm0wx";
+  };
+
+  node_modules = fetchNodeModules {
+    src = "${src}/srht";
+    nodejs = nodejs-11_x;
+    sha256 = "0axl50swhcw8llq8z2icwr4nkr5qsw2riih0a040f9wx4xiw4p6p";
+  };
+
+  patches = [
+    ./disable-npm-install.patch
+  ];
+
+  buildInputs = [
+    sassc
+    nodejs-11_x
+  ];
+
+  propagatedBuildInputs = [
+    flask
+    flask_login
+    humanize
+    sqlalchemy
+    sqlalchemy-utils
+    psycopg2
+    markdown
+    bleach
+    requests
+    beautifulsoup4
+    pgpy
+    misaka
+    pygments
+
+    # Unofficial runtime dependency?
+    alembic
+  ];
+
+  PKGVER = version;
+
+  preBuild = ''
+    cp -r ${node_modules} srht/node_modules
+  '';
+
+  # No actual? tests but seems like it needs this anyway
+  preCheck = let
+    # Just a sample config
+    config = writeText "config.ini" ''
+      [meta.sr.ht]
+      origin=http://meta.sr.ht.local
+    '';
+  in ''
+    # Tests need config option "origin" from "meta.sr.ht"
+    cp ${config} config.ini
```

Comment:
Not really someone who actively uses Python, but looking at the code [1], there is a possibility that the local `config.ini` overrides the global configuration. However, I don't see any `config.ini`s (find -name '*.ini') in the output derivation so I believe it shouldn't matter.

The login/register links in meta.sr.ht work fine for me, however, there is a "broken" link that leads to `sr.ht` instead of `/login`

[1] https://git.sr.ht/~sircmpwn/core.sr.ht/tree/master/srht/config.py#L12

Reply:
I think I was able to resolve this with service.srht.*.domain settings.